app/src/androidTest/java/com/okta/oidc/example/MfaTest.java

```diff
@@ -0,0 +1,237 @@
+import static com.okta.oidc.example.Utils.customTabInteractionWithMfa;
+import static junit.framework.TestCase.assertTrue;
+import static org.hamcrest.core.StringContains.containsString;
+
+@FixMethodOrder(MethodSorters.NAME_ASCENDING)
+@RunWith(AndroidJUnit4.class)
+@LargeTest
+public class MfaTest {
+    //app resource ids
+    private static final String ID_GET_PROFILE = "com.okta.oidc.example:id/get_profile";
+    private static final String ID_CLEAR_ALL = "com.android.systemui:id/dismiss_text";
+    private static final String OKTA_OIDC_APP = "Okta OIDC Plain";
+    private static final String OKTA_VERIFY = "com.okta.android.auth";
+    private static final String NEXUS_LAUNCHER = "com.google.android.apps.nexuslauncher";
+    private static final String SYSTEM_UI = "com.android.systemui";
+    private static final String ID_TITLE = "android:id/title";
+    private static final String CLASS_TEXTVIEW = "android.widget.TextView";
+    private static final String NOTIFICATION = "MFA Notification";
+    private static final String DENY_TEXTVIEW = "You have chosen to reject this login.";
+
+    private UiDevice mDevice;
+    @Rule
+    public ActivityTestRule<PlainActivity> activityRule = new ActivityTestRule<>(PlainActivity.class);
+
+    private UiObject getProfileButton() {
+        return mDevice.findObject(new UiSelector().resourceId(ID_GET_PROFILE));
+    }
+
+    private UiObject getSendPushButton() {
+        return mDevice.findObject(new UiSelector().text(TEXT_SEND_PUSH));
+    }
+
+    private UiObject getApproveButton() {
+        return mDevice.findObject(new UiSelector().description(TEXT_APPROVE));
+    }
+
+    private UiObject getDenyButton() {
+        return mDevice.findObject(new UiSelector().description(TEXT_DENY));
+    }
+
+    private void clearAllNotifications() {
+        mDevice.openNotification();
+        mDevice.wait(Until.hasObject(By.pkg(SYSTEM_UI)), TRANSITION_TIMEOUT);
+        UiObject2 clearAll = mDevice.findObject(By.res(ID_CLEAR_ALL));
+        if (clearAll != null) {
+            clearAll.click();
+        } else {
+            mDevice.click(mDevice.getDisplayWidth() / 2, mDevice.getDisplayHeight() / 2);
+        }
+    }
+
+    private UiObject waitForNotification() {
+        mDevice.openNotification();
+        mDevice.wait(Until.hasObject(By.pkg(SYSTEM_UI)), TRANSITION_TIMEOUT);
+        UiObject notification = mDevice.findObject(
+                new UiSelector().resourceId(ID_TITLE)
+                        .className(CLASS_TEXTVIEW)
+                        .packageName(SYSTEM_UI)
+                        .textContains(NOTIFICATION));
+        notification.waitForExists(NOTIFICATION_TIMEOUT);
+        return notification;
+    }
+
+    @Before
+    public void setUp() {
+        mDevice = UiDevice.getInstance(getInstrumentation());
+    }
+
+    @Test
+    public void signInApproveViaNotification() throws UiObjectNotFoundException {
+        clearAllNotifications();
```

Comment:
every test is doing `clearAllNotifications` so we should just do it in setup or teardown

Reply:
fixed